src/components/TextInput/BaseTextInput.js

```diff
@@ -301,7 +302,7 @@ class BaseTextInput extends Component {
                         style={[...this.props.inputStyle, styles.hiddenElementOutsideOfWindow, styles.visibilityHidden]}
                         onLayout={e => this.setState({textInputWidth: e.nativeEvent.layout.width})}
                     >
-                        {this.state.value || this.props.placeholder}
+                        {this.value || this.props.placeholder}
```

Comment:
But the updated `value` isn't reflected when calculating the width, because no re-renders happen on typing something in the input (which is a good thing). 

This can only be solved by converting `value` to a state of the component.